overlord/state/notices.go

```diff
@@ -243,7 +259,12 @@ func (s *State) AddNotice(userID *uint32, noticeType NoticeType, key string, opt
 
 	now := options.Time
 	if now.IsZero() {
-		now = time.Now()
+		now = getTimeNow()
+		// ensure that two notices never have the same sent time
+		if !now.After(s.noticeLastDate) {
+			now = s.noticeLastDate.Add(time.Nanosecond)
```

Comment:
I'm confused, does parseOptionalTime in daemon support ns precision?

Reply:
Yes. In fact, I added an specific test for that. Is the next PR.

Reply:
The microsecond problem is only in GLib. GDateTime internally limits everything to microseconds. That confused me.